azure-iot-common/tests/test_asyncio_compat.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import asyncio
+import sys
+from azure.iot.common import asyncio_compat
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+def dummy_coroutine():
+    async def coro():
+        return
+
+    return coro
+
+
+@pytest.mark.skipif(sys.version_info < (3, 7), reason="Requires Python 3.7+")
+async def test_get_running_loop(mocker, event_loop):
+    spy_get_running_loop = mocker.spy(asyncio, "get_running_loop")
+    result = asyncio_compat.get_running_loop()
+    assert result == event_loop
+    assert spy_get_running_loop.call_count == 1
+    assert spy_get_running_loop.call_args == mocker.call()
+
+
+@pytest.mark.skipif(sys.version_info >= (3, 7), reason="Requires Python 3.6 or below")
+async def test_get_running_loop_compat(mocker, event_loop):
+    spy_get_event_loop = mocker.spy(asyncio, "_get_running_loop")
+    result = asyncio_compat.get_running_loop()
+    assert result == event_loop
+    assert spy_get_event_loop.call_count == 1
+    assert spy_get_event_loop.call_args == mocker.call()
+
+
+@pytest.mark.skipif(sys.version_info >= (3, 7), reason="Requires Python 3.6 or below")
+async def test_get_running_loop_compat_no_running_loop(mocker):
```

Comment:
>test_get_running_loop_compat_no_running_loop [](start = 10, length = 44)

Why is there no 3.7 version?  If this module guarantees some behavior for all versions, why are you only testing it for some versions?

Reply:
There's no case for it in 3.7. In 3.6 we have to raise the error ourselves, thus the test. In 3.7, asyncio throws it for us. 

---
In reply to: [251153162](https://github.com/Azure/azure-iot-sdk-python-preview/pull/43#discussion_r251153162) [](ancestors = 251153162)

Reply:
I edited that comment a few times and somehow some of it got lost (and different parts of different edits got combined)

I've since added the test. My rationale had been that since I'd be forcing the error with my mock there was no actual test - I was testing test code, not actual implementation code. However, I realized that just because the underlying function call throws an exception doesn't mean ours does - proving that a raised exception bubbles up has some value.

tl;dr I added the test

---
In reply to: [251155632](https://github.com/Azure/azure-iot-sdk-python-preview/pull/43#discussion_r251155632) [](ancestors = 251155632,251153867,251153162)